### ISSUE TITLE
Backwards Compatibility Issue: 0.23+

### ISSUE THREAD

**Issue by mwillbanks** [maxnowack/signaldb#1177]
So this is a strange one... Since the release of 0.23, my project no longer works with signaldb. Reverting down to 0.22 resolves that but thus doesn't include some of the more current optimizations. It appears that the build is slightly different than previously and for whatever reason I can utilize it within my codebase but then while using some of the storage mechanisms, the library cannot be found any longer.

I've attempted to completely blow away the node_modules folder and force reinstall everything but this has not worked. I updated my tsconfig to force an alias to the proper folder and simply everything I have attempted has not worked thus far.

This comes into play while using the `oby` reactivity adapter (as the maverick one never worked for me). Anyhow, will help to debug to help to solve this but unfortunately, I don't have a very easy way of determining what exactly is the root cause. 

The compiler options of my tsconfig are as follows:
```
"compilerOptions": {
  "allowJs": true,
  "esModuleInterop": true,
  "jsx": "react-native",
  "lib": ["DOM", "ESNext"],
  "moduleResolution": "node",
  "noEmit": true,
  "resolveJsonModule": true,
  "skipLibCheck": true,
  "target": "ESNext"
}
```

**Comment by maxnowack** [comment]
I now got a similar issue in one of my projects. In a @renovatebot lockfile maintenance PR, a test with reactivity fails.
The packages that were updated in the lockfile were mostly things like `vite`, `esbuild` and `rollup` packages and dependencies. Nothing directly related to `signaldb` or `@maverick-js/signals`.
I digged don't a bit and found out, that the tests are passing if I upgrade to [`vitest@2.1.7`](https://github.com/vitest-dev/vitest/releases/tag/v2.1.7). As they reverted the switch to `vite@6` in this release, my codebase is also back to `vite@5` after the version change.
So it seems like there is something wrong with `vite@6` or the combination with `signaldb`. If you're at `vite@6` try downgrading to `vite@5` until we got the root cause and have a fix for it 🙂

**Comment by mwillbanks** [comment]
> So it seems like there is something wrong with `vite@6` or the combination with `signaldb`. If you're at `vite@6` try downgrading to `vite@5` until we got the root cause and have a fix for it 🙂

Yeah, so unfortunately, using expo / react-native which utilizes babel (ugh), so I've actually attempted quite a few things through that but have not found a way to gain the existing behavior. What is really strange is how the build files are (check the dist of 0.22 vs. 0.23+ and you'll see it really changed things around which is where I think the root cause stems from).

If you're still in need of a reproduction test case, I can certainly create one but it would appear that you might not need it at this point. I'll likely be submitting a few PR's once I can get something functional so that I can add compatibility with things like react-native as the library is quite useful but certainly the reactivity has some odd drawbacks that I've found but should be able to resolve (been having to be pretty crafty with it at times but I think you may have fixed some of those in more recent versions and I've been unable to confirm).

**Comment by maxnowack** [comment]
If you're not using vite at all, then it might be really connected to #1144 
You could check this by cloning signaldb locally, revert the mentioned PR and run `npm run build && npm link` in `signaldb/packages/signaldb`. Then link signaldb in your project with `npm link signaldb` and see if the issue goes away. 
If that's the case, don't hesitate to open a PR. The goal with #1144 was to fix #988, but [the error is still there](https://github.com/maxnowack/signaldb/issues/988#issuecomment-2488125117).

**Comment by mwillbanks** [comment]
@maxnowack so looks like part of it is that the package.json is incorrectly referencing the "main" module as the file `index.umd.js` no longer exists.

**Comment by maxnowack** [comment]
I missed the `main` field in `package.json`. This is resolved in `v0.24.3`
